ring-core/src/ring/util/response.clj

```diff
@@ -92,13 +92,22 @@
       (.exists file)
         file)))
 
-(defn- file-content-length [resp]
-  (let [file ^File (:body resp)]
-    (header resp "Content-Length" (.length file))))
+(defn- file-info [^File file]
+  (if file
+    {:content        file
+     :content-length (.length file)
+     :directory?     (.isDirectory file)
+     :last-modified  (last-modified-date file)}))
 
-(defn- file-last-modified [resp]
-  (let [file ^File (:body resp)]
-    (header resp "Last-Modified" (format-date (last-modified-date file)))))
+(defn- content-length [resp len]
+  (if len
```

Comment:
Actually, make the `:content-length` and `:last-modified` fields `nil` in the case where we don't know them. The `URLConnection` class only uses numeric values because it's dealing with primatives, AFAIK.